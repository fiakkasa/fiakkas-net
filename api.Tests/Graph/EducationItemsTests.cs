using HotChocolate.Execution;

namespace api.Tests;

public class EducationItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task EducationItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  educationItems {
    totalCount
    nodes {
      categoryId
      createdAt
      description
      href
      id
      internalId
      location
      subjects
      title
      updatedAt
      version
      timePeriod {
        end
        start
      }
      category {
        associatedCategoryTypes
        createdAt
        id
        title
        updatedAt
        version
      }
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task EducationItemById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  educationItemById(id: "RWR1Y2F0aW9uSXRlbQpnMzg4OThjNjIxNjFlNDBmMjhhOWYzOWJmMWZmNDYyMjQ=") {
    categoryId
    createdAt
    description
    href
    id
    internalId
    location
    subjects
    title
    updatedAt
    version
    timePeriod {
      end
      start
    }
    category {
      associatedCategoryTypes
      createdAt
      id
      internalId
      title
      updatedAt
      version
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
