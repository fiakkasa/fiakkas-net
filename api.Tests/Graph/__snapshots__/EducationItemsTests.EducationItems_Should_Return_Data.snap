{
  "data": {
    "educationItems": {
      "totalCount": 1,
      "nodes": [
        {
          "categoryId": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
          "createdAt": "2024-01-01T00:00:00.000Z",
          "description": "Description",
          "href": "/test",
          "id": "RWR1Y2F0aW9uSXRlbQpnMzg4OThjNjIxNjFlNDBmMjhhOWYzOWJmMWZmNDYyMjQ=",
          "internalId": "38898c62-161e-40f2-8a9f-39bf1ff46224",
          "location": "Location",
          "subjects": [
            "Subject"
          ],
          "title": "",
          "updatedAt": null,
          "version": 1,
          "timePeriod": {
            "end": null,
            "start": "2024-01-01"
          },
          "category": {
            "associatedCategoryTypes": [
              "SOFTWARE_DEVELOPMENT"
            ],
            "createdAt": "2024-01-01T00:00:00.000Z",
            "id": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
            "title": "Title",
            "updatedAt": null,
            "version": 1
          }
        }
      ]
    }
  }
}
